cdap-docs/reference-manual/source/release-notes.rst

```diff
@@ -26,19 +26,584 @@ Cask Data Application Platform Release Notes
+  Removed deprecated methods in cdap-client.
+  
+- `CDAP-3253 <https://issues.cask.co/browse/CDAP-3253>`__ -
+  Added a Stream and Dataset Widget to the CDAP-UI.
+  
+- `CDAP-3390 <https://issues.cask.co/browse/CDAP-3390>`__ -
+  Added stream views, enabling reading from a single stream using various formats and
+  schemas.
+  
+- `CDAP-3419 <https://issues.cask.co/browse/CDAP-3419>`__ -
+  HBaseQueueDebugger now shows the minimum queue event transaction write pointer both for
+  each queue and for all queues.
+  
+- `CDAP-3476 <https://issues.cask.co/browse/CDAP-3476>`__ -
+  Added a Validator Transform that can be used to validate records based on a set of
+  available validators and configured to write invalid records to an error
+  dataset.
+  
+- `CDAP-3514 <https://issues.cask.co/browse/CDAP-3514>`__ -
+  Added a Query HTTP RESTful API endpoint that retrieves a count of active queries:
+  ``<base-url>/namespaces/<namespace-id>/data/explore/queries/count``.
+  
+- `CDAP-3516 <https://issues.cask.co/browse/CDAP-3516>`__ -
+  Added a service to manage the metadata of CDAP entities.
+  
+- `CDAP-3517 <https://issues.cask.co/browse/CDAP-3517>`__ -
+  Added data stores to manage business and lineage metadata for CDAP entities.
+  
+- `CDAP-3518 <https://issues.cask.co/browse/CDAP-3518>`__ -
+  Added the publishing of metadata change notifications to Apache Kafka.
+  
+- `CDAP-3519 <https://issues.cask.co/browse/CDAP-3519>`__ -
+  Added the ability to compute lineage of a CDAP dataset or stream in a given time window.
+  
+- `CDAP-3520 <https://issues.cask.co/browse/CDAP-3520>`__ -
+  Added RESTful APIs for adding/retrieving/deleting of metadata for apps/programs/datasets/streams.
+  
+- `CDAP-3521 <https://issues.cask.co/browse/CDAP-3521>`__ -
+  Added the ability to record a dataset or stream access by a CDAP program.
+  
+- `CDAP-3522 <https://issues.cask.co/browse/CDAP-3522>`__ -
+  Added the capability to search CDAP entities based on their metadata.
+  
+- `CDAP-3523 <https://issues.cask.co/browse/CDAP-3523>`__ -
+  Added RESTful APIs for searching CDAP entities based on business metadata.
+  
+- `CDAP-3549 <https://issues.cask.co/browse/CDAP-3549>`__ -
+  Added SSH port forwarding to the CDAP virtual machine.
+  
+- `CDAP-3563 <https://issues.cask.co/browse/CDAP-3563>`__ -
+  Removed faulty and unused metrics around CDAP file resource usage.
+  
+- `CDAP-3590 <https://issues.cask.co/browse/CDAP-3590>`__ -
+  Added the ability to write to multiple sinks in ETL realtime and batch applications.
+  
+- `CDAP-3626 <https://issues.cask.co/browse/CDAP-3626>`__ -
+  For the CSV and TSV stream formats, a "mapping" setting can now be specified, mapping
+  stream event columns to schema columns.
+  
+- `CDAP-3693 <https://issues.cask.co/browse/CDAP-3693>`__ -
+  Added support for CDAP to work with HDP 2.3.
+  
+- `CDAP-3787 <https://issues.cask.co/browse/CDAP-3787>`__ -
+  Provide support for writing to Elasticsearch from batch ETL applications.
```

Comment:
elastic search was moved out into cdap-plugins repo, should remove from release notes